mediator/src/MediatorPatternClasssDiagramTemplate.puml

```diff
@@ -0,0 +1,28 @@
+@startuml
+'https://plantuml.com/class-diagram
+
+interface Mediator {
+    sendMessage(message: String): void
```

Comment:
nit: can we add sender object in second argument.

Reply:
we can have 'addUser' method as well, these 2 are bare minimum to support chat_room since user is involved in mediatorImpl class?